bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java

```diff
@@ -223,18 +218,22 @@ private BigtableOptions(
     this.clusterAdminHost = Preconditions.checkNotNull(clusterAdminHost);
     this.dataHost = Preconditions.checkNotNull(dataHost);
     this.port = port;
-    this.credential = credential;
     this.projectId = projectId;
-    this.callTimingReportPath = callTimingReportPath;
-    this.callStatusReportPath = callStatusReportPath;
     this.zone = zone;
     this.cluster = cluster;
-    this.retryOptions = retryOptions;
-    this.rpcRetryExecutorService = rpcRetryExecutorService;
     this.customEventLoopGroup = customEventLoopGroup;
     this.channelCount = channelCount;
-    this.timeoutMs = timeoutMs;
-    this.userAgent = userAgent;
+
+    ChannelOptions.Builder optionsBuilder = new ChannelOptions.Builder();
+    optionsBuilder.setCallTimingReportPath(callTimingReportPath);
+    optionsBuilder.setCallStatusReportPath(callStatusReportPath);
+    optionsBuilder.setCredential(credential);
+    optionsBuilder.setRetryOptions(retryOptions);
+    optionsBuilder.setScheduledExecutorService(rpcRetryExecutorService);
+    optionsBuilder.setChannelCount(channelCount);
+    optionsBuilder.setTimeoutMs(timeoutMs);
+    optionsBuilder.setUserAgent(userAgent);
+    this.channelOptions = optionsBuilder.build();
```

Comment:
channelCount seems to be the only parameter that is both in ChannelOptions and BigtableOptions. Asides from that the objects look quite disjoint. Maybe the BigtableOptionsBuilder should take an already built ChannelOptions instead of basically wrapping a builder into another builder. I think that would eliminate indirection and lines of code. Thoughts?